src/components/Forecast.tsx

```diff
@@ -10,42 +10,48 @@ interface Props {
 	units?: Unit
 }
 
+function useTemperatureConverter() {
+	return function(temp: number, toUnit:Unit):number{
+		switch (toUnit) {
+		      case Unit.Imperial: 
+		      	 temp=Math.floor((temp * 9/5)+32);
+				 break;
+		      default: 
+		      	 temp=Math.floor(temp + 273.15);
+				 break;
+		}
+		return temp;
+	}
+  
+}
+
 export const Forecast: React.FC<Props> = ({forecast, units}) => {
 	
-	const [newUnits,setNewUnits]=useState<string>('ºC');
+	const [tempUnits,setTempUnits]=useState<string>("ºC");
```

Comment:
The temperature units is received in the `units` from the props. Also, you initialized in Metric, but it can be received in Standard or Imperial aswell.
```suggestion
```